include/veriblock/storage/repo_batch_adaptor.hpp

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef VERIBLOCK_POP_CPP_REPO_BATCH_ADAPTOR_HPP
+#define VERIBLOCK_POP_CPP_REPO_BATCH_ADAPTOR_HPP
+
+#include "batch.hpp"
+#include "pop_storage.hpp"
+
+namespace altintegration {
+
+struct RepoBatchAdaptor : public Batch {
+  ~RepoBatchAdaptor() override = default;
+
+  RepoBatchAdaptor(PopStorage& pop) : pop_(pop) {}
+
+  bool writeBlock(const BlockIndex<BtcBlock>& value) override {
+    pop_.getBlockRepo<BlockIndex<BtcBlock>>().put(value);
+    return true;
+  }
+  bool writeBlock(const BlockIndex<VbkBlock>& value) override {
+    pop_.getBlockRepo<BlockIndex<VbkBlock>>().put(value);
+    return true;
+  }
+  bool writeBlock(const BlockIndex<AltBlock>& value) override {
+    pop_.getBlockRepo<BlockIndex<AltBlock>>().put(value);
+    return true;
+  }
+
+  bool writeTip(const BlockIndex<BtcBlock>& value) override {
+    pop_.saveTip(value);
+    return true;
+  }
+  bool writeTip(const BlockIndex<VbkBlock>& value) override {
+    pop_.saveTip(value);
+    return true;
+  }
+  bool writeTip(const BlockIndex<AltBlock>& value) override {
+    pop_.saveTip(value);
+    return true;
```

Comment:
It doesn't look like a batch at all. Is it some kind of a temporary class?

Reply:
It is not temporary. We use `flush` to actually write on disk, right? If so, it should not matter whether we use batch or no

Reply:
I do not see how the inheritance from the Batch class is being used here. All I see is a collection of PopStorage method wrappers. You could simply use PopStorage methods directly and remove this helper class completely.